freeipa/src/main/java/com/sequenceiq/freeipa/service/binduser/UserSyncBindUserService.java

```diff
@@ -0,0 +1,96 @@
+package com.sequenceiq.freeipa.service.binduser;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.freeipa.api.v1.ldap.model.describe.DescribeLdapConfigResponse;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.FreeIpaClientExceptionUtil;
+import com.sequenceiq.freeipa.client.model.User;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.ldap.LdapConfig;
+import com.sequenceiq.freeipa.ldap.LdapConfigService;
+import com.sequenceiq.freeipa.ldap.v1.LdapConfigV1Service;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+public class UserSyncBindUserService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserSyncBindUserService.class);
+
+    private static final String USERSYNC_USER_POSTFIX = "usersync";
+
+    private static final String USER_ADMINISTRATOR_ROLE = "User Administrator";
+
+    @Inject
+    private LdapConfigV1Service ldapConfigV1Service;
+
+    @Inject
+    private LdapConfigService ldapConfigService;
+
+    @Inject
+    private LdapBindUserNameProvider userNameProvider;
+
+    @Inject
+    private FreeIpaClientFactory ipaClientFactory;
+
+    public void createUserAndLdapConfig(Stack stack) throws FreeIpaClientException {
+        cleanupAlreadyExistingLdapConfig(stack.getEnvironmentCrn(), stack.getAccountId());
+        ldapConfigV1Service.createNewLdapConfig(stack.getEnvironmentCrn(), USERSYNC_USER_POSTFIX, stack, true);
+        addUserAdminRoleForBindUser(stack);
```

Comment:
It's theoretically possible for FMS to fail after adding the user but before adding the admin role. The current code will skip adding the admin role later because the user already exists. I think we should split up the creation and role assignment into two, separately checked, methods.

Reply:
I will add the role assignment check to `doesBindUserAndConfigAlreadyExist` method, so this could not happen during upgrade.
During provisioning it's not an issue, as it would be invoked again.